### PR TITLE
[FLINK-7013] [shading] Introduce flink-shaded-netty-4

### PR DESCRIPTION
**Pull request by zentol**
This PR is based on #4425.

## What is the purpose of the change

This PR integrates the shaded netty dependency from flink-shaded. Basically, replace all usages of netty with the shaded netty dependeny and remove all traves of the original dependeny.

## Brief change log

  - replace all netty/netty-router dependencies with flink-shaded-netty
  - replace all netty imports
  - modify illegal import checkstyle rule to forbid unshaded netty imports
  - add check in travis watchdog that no unshaded netty classes are present in flink-dist

## Verifying this change

- check that compilation works
- check artifacts for inclusion of shaded netty
- check artifacts for exclusion of unshaded netty
- check that we are not exposing a vanilla netty dependency with maven
- start a cluster and run some examples
- check that the web interface still works

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-dist/pom.xml [hunk 1]
```diff
@@ -204,6 +204,11 @@ under the License.
 				</exclusion>
 			</exclusions>
 		</dependency>
+		
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-shaded-netty</artifactId>
+		</dependency>
```
Comment (rmetzger) [inline flink-dist/pom.xml]: Why does this need to be added explicitly to flink-dist?

Comment (zentol) [inline flink-dist/pom.xml]: This may be a relic of a previous version where all other dependency declarations were set to `<provided>`.

Comment (rmetzger) [inline flink-dist/pom.xml]: So we are going to keep it?

Comment (zentol) [inline flink-dist/pom.xml]: I'll check whether it is necessary and remove it if it isnt.

**Comment by zentol** [comment]
@rmetzger I've removed the dependency from flink-dist; it wasn't necessary.

File: tools/travis_mvn_watchdog.sh [hunk 2]
```diff
@@ -290,6 +290,15 @@ check_shaded_artifacts() {
 		echo "=============================================================================="
 		return 1
 	fi
+
+    NETTY=`cat allClasses | grep '^io/netty' | wc -1`
```
Comment (rmetzger) [inline tools/travis_mvn_watchdog.sh]: really minor, but: Tab vs space

Comment (zentol) [inline tools/travis_mvn_watchdog.sh]: nice catch, will fix it.

**Comment by rmetzger** [comment]
+1 to merge:
- Checked with a project depending on Flink. It doesn't fail anymore after using your PR
- mvn build passes
- the dependency-red-pom of flink-runtime looks good  (no netty deps)
- start-local.sh works.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**